Add net.bontal.Catalyst

pewriebontal opened the pull request:
<!-- ⚠️⚠️  Submission pull request MUST be made against the `new-pr` **base branch** ⚠️⚠️  -->

<!-- 💡 Please go to the preview tab to view the markdown below 💡 -->

### Please confirm your submission meets all the criteria

<!-- 💡 Please replace each `[ ]` with `[X]` when the step is complete 💡 -->

<!-- 💡 Please tick and write 'N/A' with a reason if a checklist item below is not applicable 💡 -->

- [X] Please describe the application briefly. < Catalyst is an intelligent task manager that uses AI to parse natural language into structured tasks. It automatically prioritizes them using the Eisenhower Matrix (Do, Schedule, Delegate, Eliminate). The app is designed to be privacy-first with local storage and features a clean, minimalist interface. >
- [X] Please attach a video showcasing the application on Linux using the Flatpak. <[Screencast_20250702_021800.webm](https://github.com/user-attachments/assets/bafc3551-2755-4142-811e-1d79e4d8d53b)>
- [X] The Flatpak ID follows all the rules listed in the [Application ID requirements][net.bontal.Catalyst].
- [X] I have read and followed all the [Submission requirements][reqs] and the [Submission guide][reqs2] and I agree to them.
- [X] I am an _(please keep whichever is applicable and remove the rest)_ author/developer/upstream contributor to the project.
      If not, I contacted upstream developers about this submission. **Link:**

<!-- 💡 Please mention below the GitHub usernames of any additional maintainers needed (if any) 💡 -->

<!-- ⚠️⚠️  Please DO NOT modify anything below this line ⚠️⚠️  -->

[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[reqs2]: https://docs.flathub.org/docs/for-app-authors/submission

hfiguiere commented on net.bontal.Catalyst.yml:
```diff
@@ -0,0 +1,59 @@
+      - update-desktop-database /app/share/applications
+    sources:
+      - type: file
+        url: https://pkg.bontal.net/downloads/releases/catalyst-1.1.0-linux-x64.tar.gz
```
what happen with the aarch64 package?

hfiguiere commented on icon-64.png:
you don't need this. remove it.

pewriebontal replied:
ok

hfiguiere commented on icon-128.png:
you don't need this. remove it.

hfiguiere commented on net.bontal.Catalyst.yml:
```diff
@@ -0,0 +1,59 @@
+      - install -Dm644 icon-256.png /app/share/icons/hicolor/256x256/apps/net.bontal.Catalyst.png
+      - install -Dm644 icon-128.png /app/share/icons/hicolor/128x128/apps/net.bontal.Catalyst.png
+      - install -Dm644 icon-64.png /app/share/icons/hicolor/64x64/apps/net.bontal.Catalyst.png
+      - desktop-file-validate /app/share/applications/net.bontal.Catalyst.desktop
```
```suggestion
```

pewriebontal commented on net.bontal.Catalyst.yml:
```diff
@@ -0,0 +1,59 @@
+app-id: net.bontal.Catalyst
+runtime: org.gnome.Platform
+runtime-version: "48"
+sdk: org.gnome.Sdk
```
The org.gnome.Platform is required to build the zenity dependency, as it needs libadwaita. If there's a more minimal way to achieve this that you'd recommend, I'm happy to make the change.

pewriebontal commented on net.bontal.Catalyst.desktop:
noted

pewriebontal commented:
Hi @hfiguiere, thank you for all your guidance through the process.

I have pushed the final changes and believe I have addressed all the feedback:

*   **Platform:** The manifest now uses `org.freedesktop.Platform:24.08` and builds the required `libsecret` dependency, as `zenity` is already included in the runtime.
*   **Architecture:** The build is restricted to `x86_64` as aarch64 binaries are not yet available.
*   **Icons:** The smaller, redundant icon files and their corresponding install commands have been removed.
*   **.desktop File:** I've noted your feedback and added `.desktop` file directly in the upstream package.

I believe the manifest is now complete and ready for final approval. Please let me know if there is anything else I've missed.

bbhtt commented on net.bontal.Catalyst.yml:
```diff
@@ -0,0 +1,65 @@
+
+  - --share=network
+
+  - --talk-name=org.freedesktop.secrets
+
+  - --talk-name=org.freedesktop.GeoClue2
+  - --system-talk-name=org.freedesktop.GeoClue2
```
what are these 3 for?

bbhtt replied:
Why do you need both system and session for geolocation?

bbhtt replied:
> flutter_secure_storage

Does this support the secrets portal?

pewriebontal replied:
**GeoClue2 - Both permissions needed:**
I tested this and both are actually required. Without both session and system bus access, the location picker crashes when trying to get the user's current position. I believed some distros need one, some need the other, so I need both for compatibility.

**flutter_secure_storage portal support:**
Yes, I tested it and it does support the secrets portal! It works fine with `--talk-name=org.freedesktop.portal.secrets`.

bbhtt replied:
> It works fine with --talk-name=org.freedesktop.portal.secrets.

You don't need that, it is granted by default.

pewriebontal replied:
noted

bbhtt replied:
Also Geoclue doesn't run in session not sure what you are testing.

bbhtt replied:
And how did you test secrets portal support?
 I can see no mention of it in the flutter_secrets_storage repo?

bbhtt commented on net.bontal.Catalyst.yml:
```diff
@@ -0,0 +1,65 @@
+  - --socket=fallback-x11
+  - --device=dri
+
+  - --share=network
```
why do you need network?

pewriebontal replied:
The network permission is for the Al task parsing feature that connects to online APls such as Gemini or user's chosen API.

The app also works completely offline.

bbhtt commented on net.bontal.Catalyst.metainfo.xml:
```diff
@@ -0,0 +1,97 @@
+
+  <metadata_license>CC0-1.0</metadata_license>
+
+  <project_license>LicenseRef-proprietary=https://catalyst.bontal.net/license</project_license>
```
![Image](https://github.com/user-attachments/assets/9b8e2987-af91-44d8-a201-c53936c3fb48)

pewriebontal replied:
My apologies, I haven't updated the website yet, so the link is indeed still broken. I'll get the license page live by tonight SGT.

bbhtt reviewed (changes requested):
Metainfo, desktopfile and icon must be inside the deb. These can be used by distributions too.

hfiguiere commented:
> **Architecture:** The build is restricted to `x86_64` as aarch64 binaries are not yet available.

No it's not. The build will just fail.

pewriebontal commented:
> Metainfo, desktopfile and icon must be inside the deb. These can be used by distributions too.

Just to be perfectly clear about your feedback: if I include the metainfo.xml, desktopfile, and icon directly inside the application's release tarball (in their standard share/ locations), should I then:
1. Remove them as type: file entries from the Flatpak manifest's sources section?
2. And also remove those individual files from this Flatpak manifest repository?

I want to make sure I get this fully correct. Thank you!

bbhtt commented:
Yes correct.

pewriebontal commented:
> Is secret storing working now since you removed the permission? I didn't get an answer to [#6677 (comment)](https://github.com/flathub/flathub/pull/6677#discussion_r2179042502)

@bbhtt Yes, the app works perfectly fine.

bbhtt commented:
/merge head=f5b0789488d1b6cba64b335c6777a81c73377f94
